flow/src/org/labkey/flow/data/FlowProtocolSchema.java

```diff
@@ -54,6 +54,7 @@ public FlowAssayProvider getProvider()
     public ExpRunTable createRunsTable(ContainerFilter cf)
     {
         FlowSchema flowSchema = new FlowSchema(getUser(), getContainer());

```

Comment:
This class looks like it's just a wrapper for FlowSchema.  Maybe it would be clearer to follow a wrapper pattern and construct the FlowSchema once in the FlowProtocolSchema constructer and then delegate addContextualRole() and getContextualRoles().  The idea seems right.